### PR TITLE
V1.0.12 of Hotline Android SDK

### PR DESCRIPTION
**Pull request by jishnu7**
This change is [Reviewable](https://reviewable.io/reviews/hashcube/hotline/1)

### CODE REVIEW

File: android/HotlinePlugin.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    JSONObject reqJson;
+    Map<String, String> userMeta = new HashMap<String, String>();
+
+    try{
+      reqJson = new JSONObject(param);
+      userMeta.put(reqJson.getString("field_name"), reqJson.getString("value"));
+    } catch (Exception e){
+      logger.log(TAG + "{exception}", "" + e.getMessage());
+    }
+    Hotline.getInstance(appContext).updateUserProperties(userMeta);
```
Comment (bijoshtj) [inline android/HotlinePlugin.java:135]: wont this throw err if exception occured in above line?

Comment (jishnu7) [inline android/HotlinePlugin.java:135]: No. It will just pass an empty userMeta to hotline.

**Comment by rampr** [comment]
Reviewed 141 of 471 files at r1.
Review status: 141 of 471 files reviewed at latest revision, 3 unresolved discussions.

---

_[package.json, line 3 [r1]](https://reviewable.io:443/reviews/hashcube/hotline/1#-KPk_kDgYDhKtaATaJA7:-KPk_kDgYDhKtaATaJA8:b-9pxma2) ([raw file](https://github.com/hashcube/hotline/blob/f2532822b2f6aa4843d58d33735aeb2162002c83/package.json#L3)):_

> ``` JSON
> {
>   "name": "hotline",
>   "version": "1.0.12-1",
> ```

is the version value okay?

---

_[android/manifest.xml, line 7 [r1]](https://reviewable.io:443/reviews/hashcube/hotline/1#-KPkbM_eYn8VIs8BPcyI:-KPkbM_eYn8VIs8BPcyJ:b-m7fhju) ([raw file](https://github.com/hashcube/hotline/blob/f2532822b2f6aa4843d58d33735aeb2162002c83/android/manifest.xml#L7)):_

> ``` XML
> 
>   
>   <uses-permission android:name="com.google.android.c2dm.permission.RECEIVE" />
> ```

Shouldn't we add the GCM jar file from google so it receives pushes?

That way this plugin will work independently of onesignal added in the build

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/hotline/1)_

**Comment by rampr** [comment]
Reviewed 5 of 471 files at r1.
Review status: 146 of 471 files reviewed at latest revision, 3 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/hotline/1)_

**Comment by rampr** [comment]
Reviewed 325 of 471 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/hotline/1)_

**Comment by jishnu7** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

_[package.json, line 3 [r1]](https://reviewable.io:443/reviews/hashcube/hotline/1#-KPk_kDgYDhKtaATaJA7:-KPkelEApTRqWIvM-XAa:box3cl0) ([raw file](https://github.com/hashcube/hotline/blob/f2532822b2f6aa4843d58d33735aeb2162002c83/package.json#L3)):_
<details><summary>

<i>Previously, rampr (ram) wrote…</i></summary>



> is the version value okay?
> </details>
> Yes. But I'll change this.

It can happen like Hotline android sdk and iOS sdk have different versions. It will be better to have our own version.

---

_[android/manifest.xml, line 7 [r1]](https://reviewable.io:443/reviews/hashcube/hotline/1#-KPkbM_eYn8VIs8BPcyI:-KPkfSCsQ2yWrYjxYgUs:bwyvlkt) ([raw file](https://github.com/hashcube/hotline/blob/f2532822b2f6aa4843d58d33735aeb2162002c83/android/manifest.xml#L7)):_
<details><summary>

<i>Previously, rampr (ram) wrote…</i></summary>



> Shouldn't we add the GCM jar file from google so it receives pushes?
> 
> That way this plugin will work independently of onesignal added in the build
> </details>
> Not adding GCM right now as Hotline doesn't handle/process the messages automatically, like onesignal does. 

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/hotline/1)_

**Comment by rampr** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_[package.json, line 3 [r1]](https://reviewable.io:443/reviews/hashcube/hotline/1#-KPk_kDgYDhKtaATaJA7:-KPkipLxmD29lNF3wHcl:b-1opiju) ([raw file](https://github.com/hashcube/hotline/blob/f2532822b2f6aa4843d58d33735aeb2162002c83/package.json#L3)):_
<details><summary>

<i>Previously, jishnu7 (Jishnu Mohan) wrote…</i></summary>



> Yes. But I'll change this.
> 
> It can happen like Hotline android sdk and iOS sdk have different versions. It will be better to have our own version.
> 
> </details>
> yeah, better to maintain our own version

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/hotline/1)_

**Comment by jishnu7** [comment]
Review status: 469 of 471 files reviewed at latest revision, 2 unresolved discussions.

---

_[package.json, line 3 [r1]](https://reviewable.io:443/reviews/hashcube/hotline/1#-KPk_kDgYDhKtaATaJA7:-KPkjX0o8DC6ozUaM89_:b-896fix) ([raw file](https://github.com/hashcube/hotline/blob/f2532822b2f6aa4843d58d33735aeb2162002c83/package.json#L3)):_
<details><summary>

<i>Previously, rampr (ram) wrote…</i></summary>



> yeah, better to maintain our own version
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/hotline/1)_

**Comment by rampr** [comment]
Reviewed 1 of 1 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/hotline/1)_

**Comment by rampr** [comment]
LGTM

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/hotline/1#-:-KPkkSHDve3ObcgE4gFp:bb6ywmx)_

**Comment by rohanhc** [comment]
LGTM

**Comment by bijoshtj** [comment]
Reviewed 471 of 471 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

_[js/index.js, line 18 [r1]](https://reviewable.io:443/reviews/hashcube/hotline/1#-KPkgq7MLrLCIIt0pLug:-KPkgq7MLrLCIIt0pLuh:bftcv6y) ([raw file](https://github.com/hashcube/hotline/blob/f2532822b2f6aa4843d58d33735aeb2162002c83/js/index.js#L18)):_

> ``` JavaScript
>   "use strict";
> 
>   // Pop off the first two arguments and keep the rest
> ```

isnt it poping of first arg only!!

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/hotline/1)_

**Comment by jishnu7** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_[js/index.js, line 18 [r1]](https://reviewable.io:443/reviews/hashcube/hotline/1#-KPkgq7MLrLCIIt0pLug:-KPpeFpt2VIDFZ-wquoP:bkomwrg) ([raw file](https://github.com/hashcube/hotline/blob/f2532822b2f6aa4843d58d33735aeb2162002c83/js/index.js#L18)):_
<details><summary>

<i>Previously, Bijoshtj (Bijosh T J) wrote…</i></summary>



> isnt it poping of first arg only!!
> </details>
> Let me check. I just copied the function from mobihelp sdk

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/hotline/1)_

**Comment by jishnu7** [comment]
Review status: 470 of 471 files reviewed at latest revision, 2 unresolved discussions.

---

_[js/index.js, line 18 [r1]](https://reviewable.io:443/reviews/hashcube/hotline/1#-KPkgq7MLrLCIIt0pLug:-KPpeqIaNg62shFzNK4f:b-896fix) ([raw file](https://github.com/hashcube/hotline/blob/f2532822b2f6aa4843d58d33735aeb2162002c83/js/index.js#L18)):_
<details><summary>

<i>Previously, jishnu7 (Jishnu Mohan) wrote…</i></summary>



> Let me check. I just copied the function from mobihelp sdk
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/hotline/1)_

**Comment by rampr** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/hotline/1)_

**Comment by bijoshtj** [comment]
Reviewed 1 of 1 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/hotline/1)_

**Comment by bijoshtj** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hashcube/hotline/1)_

**Pull request merged**